scanner/keppel/main.go

```diff
@@ -4,9 +4,125 @@
 package main
 
 import (
-	"fmt"
+	"os"
+	"sync"
+
+	"github.com/cloudoperators/heureka/scanners/keppel/models"
+	"github.com/cloudoperators/heureka/scanners/keppel/processor"
+	"github.com/cloudoperators/heureka/scanners/keppel/scanner"
+	"github.com/kelseyhightower/envconfig"
+	log "github.com/sirupsen/logrus"
 )
 
+func init() {
+	// Log as JSON instead of the default ASCII formatter.
+	log.SetFormatter(&log.JSONFormatter{})
+
+	// Output to stdout instead of the default stderr
+	// Can be any io.Writer, see below for File example
+	log.SetOutput(os.Stdout)
+
+	// Only log the warning severity or above.
+	log.SetLevel(log.DebugLevel)
+}
+
 func main() {
-	fmt.Println("This is the Keppel scanner")
+	var wg sync.WaitGroup
+	var scannerCfg scanner.Config
+	err := envconfig.Process("heureka", &scannerCfg)
+	if err != nil {
+		log.WithError(err).Fatal("Error while reading env config for scanner")
+	}
+
+	var processorCfg processor.Config
+	err = envconfig.Process("heureka", &processorCfg)
+	if err != nil {
+		log.WithError(err).Fatal("Error while reading env config for processor")
+	}
+
+	keppelScanner := scanner.NewScanner(scannerCfg)
+	keppelProcessor := processor.NewProcessor(processorCfg)
+
+	err = keppelScanner.Setup()
+	if err != nil {
+		log.WithError(err).Fatal("Error during scanner setup")
+	}
+	accounts, err := keppelScanner.ListAccounts()
+	if err != nil {
+		log.WithError(err).Fatal("Error during ListAccounts")
+	}
+
+	wg.Add(len(accounts))
+
+	for _, account := range accounts {
+		go ProcessAccount(account, keppelScanner, keppelProcessor, &wg)
+	}
+
+	wg.Wait()
+}
+
+func ProcessAccount(account models.Account, keppelScanner *scanner.Scanner, keppelProcessor *processor.Processor, wg *sync.WaitGroup) error {
```

Comment:
We might consider refactoring this function into smaller ones (processRepositories, processManifests etc.) as it's already huge with lots of for-loops.